### PR TITLE
Create initial spec for RPM packaging

### PR DESCRIPTION
**Pull request by mangelajo**
This can be built with the following commands for a release (download the source from github)
```
$ spectool -g -R microshift.spec
$ rpmbuild -ba microshift.spec
```

And with the following commands for local development and changes:

```
# build an rpm for your local architecture and distro
$ make devel-rpm

# build an srpm which you can upload to a builder service
$ make devel-srpm
```
Related-Issue: #278

Signed-off-by: Miguel Angel Ajo <majopela@redhat.com>

### CODE REVIEW

File: microshift.spec [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+%ifarch x86_64
+GOARCH=amd64
+%endif
+
+make _build_local GOOS=${GOOS} GOARCH=${GOARCH}
+cp ./_output/bin/${GOOS}_${GOARCH}/microshift ./_output/microshift
+
+%install
+install -d %{buildroot}%{_bindir}
+install -p -m 755 ./_output/microshift %{buildroot}%{_bindir}/microshift
```
Comment (rootfs) [inline microshift.spec:63]: can you also add selinux modules here? or it goes to another rpm?

File: microshift.spec [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+# golang specifics
+%global golang_version 1.15
+#debuginfo not supported with Go
+%global debug_package %{nil}
+# modifying the Go binaries breaks the DWARF debugging
+%global __os_install_post %{_rpmconfigdir}/brp-compress
+
+Name: microshift
+Version: %{version}
+Release: %{release}%{dist}
```
Comment (rootfs) [inline microshift.spec:14]: where is `dist` defined?

File: microshift.spec (hunk 1 above)
Comment (mangelajo) [reply microshift.spec:63]: Yes, that was my plan as per conversation in https://github.com/redhat-et/microshift/issues/277#issuecomment-922700554

I can do that if @cooktheryan is ok :)

File: microshift.spec (hunk 2 above)
Comment (mangelajo) [reply microshift.spec:14]: It comes from rpmbuild, normally the distro where you are building, or, a mock envionment created by the builder that you request for a specific distro.

I passed the output SRPM from this, and https://copr.fedorainfracloud.org/coprs/mangelajo/microshift/build/2829853/


I was thinking that we could automate that for every tag, and then have one for every new git commit in the form of "devel" version and "%{timestamp}.git%{commit-sha}" release.

**Comment by mangelajo** [comment]
Thanks for the eyes @rootfs

File: microshift.spec (hunk 2 above)
Comment (rootfs) [reply microshift.spec:14]: i see, thanks for the education.

**Comment by rootfs** [comment]
/lgtm
(good for me, waiting for @cooktheryan for selinux and @copejon on build decisions)

**Comment by mangelajo** [comment]
Probably we should also add dependencies (cri-o cri-tools)

**Comment by cooktheryan** [comment]
@mangelajo feel free to package them together. I can get you the required steps to generate the .pp file if you need them. Most items you should be able to derive from the install.sh

**Pull request merged**